contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/zonal/ZonalIdleVehicleXYVisualiser.java

```diff
@@ -66,37 +75,34 @@ private void initEntryMap() {
 	@Override
 	public void handleEvent(ActivityStartEvent event) {
 		if (event.getActType().equals(DrtActionCreator.DRT_STAY_NAME)) {
-			String zone = zonalSystem.getZoneForLinkId(event.getLinkId());
-			if (zone != null) {
-				Integer oldNrOfVeh = zoneEntries.get(zone).getLast().getSecond();
-				zoneEntries.get(zone).add(new Tuple<>(event.getTime(), oldNrOfVeh + 1));
+			if(this.fleet.getVehicleSpecifications().containsKey(Id.create(event.getPersonId().toString(), DvrpVehicle.class))){
+				String zone = zonalSystem.getZoneForLinkId(event.getLinkId());
+				if (zone != null) {
+					LinkedList<Tuple<Double, Integer>> zoneTuples = zoneEntries.get(zone);
+					Integer oldNrOfVeh = zoneTuples.getLast().getSecond();
+					zoneTuples.add(new Tuple<>(event.getTime(), oldNrOfVeh + 1));
+				}
 			}
 		}
-
-		if (event.getActType().equals(VrpAgentLogic.AFTER_SCHEDULE_ACTIVITY_TYPE)) {
-			String zone = zonalSystem.getZoneForLinkId(event.getLinkId());
-			if (zone != null) {
-				Integer oldNrOfVeh = zoneEntries.get(zone).getLast().getSecond();
-				zoneEntries.get(zone).add(new Tuple<>(event.getTime(), oldNrOfVeh - 1));
-			}
-		}
-
 	}
 
 	@Override
 	public void handleEvent(ActivityEndEvent event) {
 		if (event.getActType().equals(DrtActionCreator.DRT_STAY_NAME)) {
-			String zone = zonalSystem.getZoneForLinkId(event.getLinkId());
-			if (zone != null) {
-				Integer oldNrOfVeh = zoneEntries.get(zone).getLast().getSecond();
-				zoneEntries.get(zone).add(new Tuple<>(event.getTime(), oldNrOfVeh - 1));
+			if(this.fleet.getVehicleSpecifications().containsKey(Id.create(event.getPersonId().toString(), DvrpVehicle.class))){
```

Comment:
I decided to still use ActivityEnd and ActivityStart and do the well-known-but-gross id-lookup via conversion of person and vehicle id...